sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -100,7 +103,24 @@ private void init() {
             logger);
   }
 
-  public synchronized void start() {
+  public synchronized void start(final @NotNull TracesSampler tracesSampler) {
```

Comment:
kind of unrelated to this PR but we now have AuotCloseableReentrantLock that we use everywhere instead of synchronized, perhaps you'd want to switch over to it

Reply:
I'll do it in another PR, thanks for raising it!